tests/upgrades/test_contentview.py

```diff
@@ -279,3 +297,223 @@ def test_post_user_scenario_bug_1429201(self):
+            'content-type': 'yum',
+            'url': FAKE_1_YUM_REPO})
+        Repository.synchronize({'id': self.yum_repo1['id']})
+        self.module = {'name': self.puppet_module_name, 'version': '3.3.3'}
+        self.puppet_repo = make_repository({
+            'name': self.puppet_repo_name,
+            'content-type': 'puppet',
+            'product-id': self.product['id'],
+            'url': CUSTOM_PUPPET_REPO,
+        })
+        Repository.synchronize({'id': self.puppet_repo['id']})
+        self.puppet_module = PuppetModule.list({
+            'search': 'name={name} and version={version}'.format(**self.module)})[0]
+
+    def make_file_repository_upload_contents(self, options=None):
+        """Makes a new File repository, Upload File/Multiple Files
+        and asserts its success.
+        """
+        if options is None:
+            options = {
+                'name': self.file_repo_name,
+                'product-id': self.product['id'],
+                'content-type': 'file'
+            }
+        if not options.get('content-type'):
+            raise CLIFactoryError('Please provide a valid Content Type.')
+        file_repo = make_repository(options)
+        remote_path = "/tmp/{0}".format(RPM_TO_UPLOAD)
+        if 'multi_upload' not in options or not options['multi_upload']:
+            ssh.upload_file(
+                local_file=get_data_file(RPM_TO_UPLOAD),
+                remote_file=remote_path
+            )
+        else:
+            remote_path = "/tmp/{}/".format(gen_string('alpha'))
+            ssh.upload_files(local_dir=os.getcwd() + "/../data/",
+                             remote_dir=remote_path)
+
+        result = Repository.upload_content({
+            'name': file_repo['name'],
+            'organization': file_repo['organization'],
+            'path': remote_path,
+            'product-id': file_repo['product']['id'],
+        })
+        self.assertIn(
+            "Successfully uploaded file '{0}'".format(RPM_TO_UPLOAD),
+            result[0]['message'],
+        )
+        file_repo = Repository.info({'id': file_repo['id']})
+        self.assertGreater(int(file_repo['content-counts']['files']), 0)
+        return file_repo
+
+    @pre_upgrade
+    def test_cv_preupgrade_scenario(self):
+        """This is pre-upgrade scenario test to verify if we can create a
+         content-view with various repositories.
+
+         :id: a4ebbfa1-106a-4962-9c7c-082833879ae8
+
+         :steps:
+           1. Create custome yum, puppet and file repositories.
+           2. Create content-view.
+           3. Add yum, file repositories and puppet module to content view.
+           4. Publish content-view.
+
+         :expectedresults: content-view created with various repositories.
+         """
+        self.setupScenario()  # Creates custome yum, puppet repositories.
+        # Makes a new File repository
+        file_repo = self.make_file_repository_upload_contents()
```

Comment:
" # Creates custome yum, puppet repositories.
        # Makes a new File repository"

Above docstring not requires as it is expected to done if self.setupScenario() run.